src/mono/mono/mini/aot-compiler.c

```diff
@@ -9828,8 +9828,9 @@ compile_method (MonoAotCompile *acfg, MonoMethod *method)
 	mono_atomic_inc_i32 (&acfg->stats.ccount);
 
 	if (acfg->aot_opts.compiled_methods_outfile && acfg->compiled_methods_outfile != NULL) {
-		if (!mono_method_is_generic_impl (method) && method->token != 0)
-			fprintf (acfg->compiled_methods_outfile, "%x\n", method->token);
+		if (!mono_method_is_generic_impl (method) && method->token != 0) {
+			fprintf (acfg->compiled_methods_outfile, "%d\n", method->token);
```

Comment:
Why decimal.  Tokens are usually hex.  I would do `0x%08x` so it's always printed out as `0x0900abcd` etc

Reply:
Because this API returns decimal.  MetadataTokens.GetToken (https://learn.microsoft.com/en-us/dotnet/api/system.reflection.metadata.ecma335.metadatatokens.gettoken?view=net-8.0). And I need to compare what I logged here with the return of that API. It doesn't make sense to do extra conversion.

Reply:
The API returns an `int`.  How you read/write that int to the txt file doesn't matter to the API.  I think we should use hex like all the other tools that work with metadata tokens.

All you need to do is change the printf here to `0x%08x`. and then do the conversion when you read it in using `Convert.ToInt32(line, 16);`

Reply:
Sure, I could change it back to hex.